### ISSUE TITLE
`Cards in navigable grid` example is missing labeling 

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#14112]
This ticket is output of the user testing.

Example: Cards in navigable grid

Repro steps:
NVDA or JAWS is in application mode.
Navigate into the card in the example
Observe narration
Actual: "user card"
Expected:  it narrate some user card meaningful name + 'user card'
